app/main.py

```diff
@@ -1,8 +1,40 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        if "wife" in Person.people:
+            self.wife = None
+        if "husband" in Person.people:
+            self.husband = None
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    @classmethod
+    def reset_people(cls) -> None:
+        cls.people.clear()
+
+
+def create_person_list(people_data: list[dict]) -> list[Person]:
+
+    Person.reset_people()
+
+    person_instances = [
+        Person(data["name"], data["age"]) for data in people_data
+    ]
+
+    for data in people_data:
+        current_person = Person.people[data["name"]]
+        spouse_role = "wife" if "wife" in data else "husband"
+        spouse_name = data.get(spouse_role)
```

Comment:
The logic here assumes that each person can have only one spouse role ('wife' or 'husband'). Ensure that the input data aligns with this assumption, or adjust the logic to handle cases where both roles might be present.